staging/src/k8s.io/apiserver/pkg/registry/rest/update.go

```diff
@@ -40,8 +40,6 @@ type RESTUpdateStrategy interface {
 	runtime.ObjectTyper
 	// NamespaceScoped returns true if the object must be within a namespace.
 	NamespaceScoped() bool
-	// AllowCreateOnUpdate returns true if the object can be created by a PUT.
-	AllowCreateOnUpdate() bool
```

Comment:
I don't mind getting rid of this, but this is about the PUT endpoint. I didn't intend to change the way the PUT endpoint behaves yet. There's no reason to treat this as saying anything about PATCH, which doesn't currently support creation.

Reply:
To allow create on patch, and to use the existing storage.Update() function, I think we either need to:

- Remove AllowCreateOnUpdate(), this changes PUT behavior
- Change the signature of the Update() function to allow the value AllowCreateOnUpdate part of the strategy to be ignored, doesn't change PUT behavior, adds a field to storage.Update
- Wrap the UpdateStrategy in the patch handler so it will always return true for AllowCreateOnUpdate, also doesn't change PUT behavior, but seems really hacky

Reply:
After seeing the rest of the PR, I will admit that it's pretty dang convenient to just call this code (if it works). Was there a reason it was never rolled out everywhere?